dispatches/models/fossil_case/concrete_tube/heat_exchanger_tube.py

```diff
@@ -0,0 +1,503 @@
+
+        self.tube.add_energy_balances(
+            balance_type=self.config.tube_side.energy_balance_type,
+            has_heat_transfer=True,
+        )
+
+        self.tube.add_momentum_balances(
+            balance_type=self.config.tube_side.momentum_balance_type,
+            has_pressure_change=self.config.tube_side.has_pressure_change,
+        )
+
+        self.tube.apply_transformation()
+
+        # Add Ports for tube side
+        self.add_inlet_port(name="tube_inlet", block=self.tube)
+        self.add_outlet_port(name="tube_outlet", block=self.tube)
+
+        # Add reference to control volume geometry
+        add_object_reference(self, "tube_area", self.tube.area)
+        add_object_reference(self, "tube_length", self.tube.length)
+
+        self._make_performance()
+
+    def _make_performance(self):
+        """
+        Constraints for unit model.
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+        tube_units = self.config.tube_side.property_package.\
+            get_metadata().get_derived_units
+
+        self.d_tube_outer = Var(initialize=0.011,
+                                doc="Outer diameter of tube",
+                                units=tube_units("length"))
+        self.d_tube_inner = Var(initialize=0.010,
+                                doc="Inner diameter of tube",
+                                units=tube_units("length"))
+
+        self.tube_heat_transfer_coefficient = Var(
+            self.flowsheet().config.time,
+            self.tube.length_domain,
+            initialize=50,
+            doc="Heat transfer coefficient",
+            units=tube_units("heat_transfer_coefficient")
+        )
+
+        self.temperature_wall = Var(
+            self.flowsheet().config.time,
+            self.tube.length_domain,
+            initialize=298.15,
+            units=tube_units("temperature")
+        )
+
+        # Energy transfer between tube wall and tube
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.tube.length_domain,
+            doc="Convective heat transfer",
+        )
+        def tube_heat_transfer_eq(self, t, x):
+            return self.tube.heat[t, x] == self.tube_heat_transfer_coefficient[
+                t, x
+            ] * c.pi * pyunits.convert(self.d_tube_inner,
+                                       to_units=tube_units("length")) * (
+                pyunits.convert(self.temperature_wall[t, x],
+                                to_units=tube_units('temperature')) -
+                self.tube.properties[t, x].temperature
+            )
+
+        # Define tube area in terms of tube diameter
+        self.area_calc_tube = Constraint(
+            expr=4 * self.tube_area == c.pi * pyunits.convert(
+                self.d_tube_inner, to_units=tube_units("length"))**2
+        )
+
+    def initialize(
+        blk,
+        tube_state_args=None,
+        outlvl=idaeslog.NOTSET,
+        solver="ipopt",
```

Comment:
We recently updated how we create solver objects in initialization routines. Here, the arguments should be `solver=None, optarg={}`.

Reply:
A discussion with @jghouse88 pointed out that this would depend on using the current `main` branch of IDAES. If we are using the most recent release, then we should hold off on this.

Reply:
Also, the default for `optarg` is now `None` if we are using the latest IDAES.